encode/av1encode.c

```diff
@@ -2899,6 +2903,8 @@ static int calc_PSNR(double *psnr)
             srcyuv_ptr = mmap(0, fourM, PROT_READ, MAP_SHARED, fileno(srcyuv_fp), i);
             recyuv_ptr = mmap(0, fourM, PROT_READ, MAP_SHARED, fileno(recyuv_fp), i);
             if ((srcyuv_ptr == MAP_FAILED) || (recyuv_ptr == MAP_FAILED)) {
+                free(srcyuv_ptr);
```

Comment:
`free` will survive passing NULL, but I don't think it will survive passing MAP_FAILED which is -1. This code will most likely crash.

Reply:
use the following code to replace free, thanks!
`if (srcyuv_ptr) 
          munmap(srcyuv_ptr, fourM);
 if (recyuv_ptr) 
          munmap(recyuv_ptr, fourM);`